### PR TITLE
Add service map support

### PR DESCRIPTION
**Pull request by aocenas**
Adds support for service map requests that can be visualized in new Node Graph component (PR: https://github.com/grafana/grafana/pull/29706) 

Screenshot from 2021-01-18 18-26-28

### CODE REVIEW

File: src/DataSource.ts [hunk 1]
```diff
@@ -204,39 +211,223 @@ function parseTraceResponse(response: DataFrame): DataFrame {
...
+        title: 'All Traces',
+        url: '',
+        internal: {
+          query: {
+            ...(query || {}),
+            queryType: XrayQueryType.getTraceSummaries,
+            query: serviceQuery,
+          },
+          datasourceUid: instanceSettings.uid,
+          // @ts-ignore
```
Comment (torkelo) [inline src/DataSource.ts:376]: why the need for ts-ignore for this on?

Comment (aocenas) [reply src/DataSource.ts:376]: I just have to update the grafana-data dependency I think we still uses the 7.2 version of those here.

File: src/utils/transform.ts [hunk 2]
```diff
@@ -197,3 +216,293 @@ function valueToTag(key: string, value: string | number | undefined, type: strin
...
+        config: { unit: 't/min' },
+        labels: { NodeGraphValueType: 'secondaryStat' },
+      };
+
+  const successField = {
+    name: 'success',
+    type: FieldType.number,
+    values: new ArrayVector(),
+    labels: { NodeGraphValueType: 'arc' },
+    config: { color: { fixedColor: 'rgb(80, 171, 113)' } },
```
Comment (torkelo) [inline src/utils/transform.ts:283]: Is there a way to use named color instead: https://github.com/grafana/grafana/blob/master/packages/grafana-data/src/utils/namedColorsPalette.ts 

they adapt slightly a bit for light & dark theme

Comment (aocenas) [reply src/utils/transform.ts:283]: Yes I can map it, these are taken directly from x-ray console.

File: src/utils/transform.ts [hunk 3]
```diff
@@ -197,3 +216,283 @@ function valueToTag(key: string, value: string | number | undefined, type: strin
...
+
+    if (showRequestCounts) {
+      const count = edge.ResponseTimeHistogram.reduce((acc, h) => acc + h.Count, 0);
+      edgeSecondaryStatField.values.add(count + ' Request' + (count > 1 ? 's' : ''));
+    } else {
+      edgeSecondaryStatField.values.add(tracesPerMinute(edge.SummaryStatistics, edge.StartTime, edge.EndTime));
+    }
+  }
+
+  for (let i = 0; i < edgeTargetField.values.length; i++) {}
```
Comment (zoltanbedi) [inline src/utils/transform.ts:424]: I think this is not needed?

**Review by zoltanbedi — COMMENTED** [review]
I saw that there are multiple todo comments in the code, are you going to do them now?

File: src/utils/transform.ts (hunk 3 above)
Comment (aocenas) [reply src/utils/transform.ts:424]: Yeah does not seem like the most important part of the logic :D

**Comment by aocenas** [comment]
> I saw that there are multiple todo comments in the code, are you going to do them now?

Not all tbh. I tend to sprinkle TODO comments here and there for future reference, like possible refactor, perf improvements etc so usually they are not support important to do right away. Probably makes harder to review and figure out whether it's missed right away or not. Maybe I can use something like TODO(pr) to mark things that should be done before merging the PR. Here I guess `// TODO: needs new grafana/data` for example is somthing I want to do for this PR.

**Review by zoltanbedi — APPROVED** [review]
lgtm 🍖

**Pull request merged**